ipn/ipnlocal/serve.go

```diff
@@ -257,6 +258,165 @@ func (b *LocalBackend) ServeConfig() ipn.ServeConfigView {
 	return b.serveConfig
 }
 
+// StreamServe opens a stream to write any incoming connections made
+// to the given HostPort out to the listening io.Writer.
+//
+// If Serve and Funnel were not already enabled for the HostPort in the ServeConfig,
+// the backend enables it for the duration of the context's lifespan and
+// then turns it back off once the context is closed. If either are already enabled,
+// then they remain that way but logs are still streamed
+func (b *LocalBackend) StreamServe(ctx context.Context, w io.Writer, req ipn.ServeStreamRequest) (err error) {
+	f, ok := w.(http.Flusher)
+	if !ok {
+		return errors.New("writer not a flusher")
+	}
+	f.Flush()
```

Comment:
If this is required, put it in the signature instead of `io.Writer`?

Reply:
@bradfitz the function needs _both_ `io.Writer` and `http.Flusher` -- there isn't an interface that combines them. Should we define: 

```
type WriteFlusher interface {
  io.Writer
  http.Flusher
}
```

?

Note it would need to be exported given the package structure.